Sources/BrowserServicesKit/Suggestions/Suggestion.swift

```diff
@@ -23,8 +23,64 @@ public enum Suggestion: Equatable {
     case phrase(phrase: String)
     case website(url: URL)
     case bookmark(title: String, url: URL, isFavorite: Bool)
+    case historyEntry(title: String?, url: URL)
     case unknown(value: String)
 
+    var url: URL? {
+        get {
+            switch self {
+            case .website(url: let url),
+                 .historyEntry(title: _, url: let url),
+                 .bookmark(title: _, url: let url, isFavorite: _):
+                return url
+            case .phrase, .unknown:
+                return nil
+            }
+        }
+
+        set {
+            guard let newValue = newValue else {
+                return
+            }
+            switch self {
+            case .website:
+                self = .website(url: newValue)
+            case .historyEntry(title: let title, url: _):
+                self = .historyEntry(title: title, url: newValue)
+            case .bookmark(title: let title, url: _, isFavorite: let isFavorite):
+                self = .bookmark(title: title, url: newValue, isFavorite: isFavorite)
+            case .phrase, .unknown:
+                return
+            }
+        }
+    }
+
+    var title: String? {
+        get {
+            switch self {
+            case .historyEntry(title: let title, url: _):
+                return title
+            case .bookmark(title: let title, url: _, isFavorite: _):
+                return title
+            case .phrase, .website,.unknown:
+                return nil
+            }
+        }
+
+        set {
+            switch self {
+            case .historyEntry(title: _, url: let url):
+                self = .historyEntry(title: newValue, url: url)
+            case .bookmark(title: _, url: let url, isFavorite: let isFavorite):
+                if let newTitle = newValue {
+                    self = .bookmark(title: newTitle, url: url, isFavorite: isFavorite)
+                }
+            case .phrase, .website,.unknown:
+                return
+            }
+        }
```

Comment:
the setter seems only used in Tests

Reply:
Used in previous versions, not necessary 👍